tools/killswitch/src/killswitch.rs

```diff
@@ -0,0 +1,258 @@
+use crate::{
+    errors::{is_error, take_error, Error},
+    settings::Settings,
+    Args, Message, Result,
+};
+use futures_util::future::join_all;
+use nomad_base::{ChainSetup, ChainSetupType, ConnectionManagers, Homes};
+use nomad_core::{
+    Common, ConnectionManager, FailureNotification, Home, SignedFailureNotification, Signers,
+    TxOutcome,
+};
+use nomad_xyz_configuration::AgentSecrets;
+
+/// Main `KillSwitch` struct
+pub(crate) struct KillSwitch {
+    /// A vector of all `ChannelKiller`
+    channel_killers: Vec<ChannelKiller>,
+}
+
+/// The set of origin->destination networks
+#[derive(Debug, Clone)]
+struct Channel {
+    /// Origin network
+    home: String,
+    /// Destination network
+    replica: String,
+}
+
+/// The channel and contracts required, or errors encountered
+struct ChannelKiller {
+    /// The channel we want to kill
+    channel: Channel,
+    /// Home contract or encountered error
+    home_contract: Option<Result<Homes>>,
+    /// Connection manager or encountered error
+    connection_manager: Option<Result<ConnectionManagers>>,
+    /// Attestation signer or encountered error
+    attestation_signer: Option<Result<Signers>>,
+}
+
+impl ChannelKiller {
+    /// Have we collected *any* errors
+    fn has_errors(&self) -> bool {
+        is_error!(self.home_contract)
+            || is_error!(self.connection_manager)
+            || is_error!(self.attestation_signer)
+    }
+
+    /// Take all available errors
+    fn take_all_errors(&mut self) -> Vec<Error> {
+        vec![
+            take_error!(self.home_contract),
+            take_error!(self.connection_manager),
+            take_error!(self.attestation_signer),
+        ]
+        .into_iter()
+        .flatten()
+        .collect()
+    }
```

Comment:
The mix of options and results is a bit confusing as per a comment below about maybe just using `Result<T>`. If possible, think simplifying this so we don't have overhead of Results nested in Options would make things more understandable

Reply:
These macros aren't necessary but "iterating" over struct members like this leads to a lot of boilerplate. As for the options -- see comment above.